src/libutil/file-system.cc

```diff
@@ -641,6 +637,13 @@ std::pair<AutoCloseFD, Path> createTempFile(const Path & prefix)
     return {std::move(fd), tmpl};
 }
 
+Path makeTempPath(const Path & root, const Path & suffix)
```

Comment:
This looks pretty similar to `tempName()` in `file-system.cc` (which already uses an atomic counter)?

Reply:
I thought it seemed familiar.

Reply:
I've consolidated the two functions now.

Reply:
The original `tempName` did `%1%/%2%-%3%-%4%` whereas this one does `%1%%2%-%3%-%4%`